app/src/main/java/org/mozilla/fenix/components/metrics/GleanMetricsService.kt

```diff
@@ -468,19 +472,21 @@ class GleanMetricsService(private val context: Context) : MetricsService {
             adjustCampaign.set(context.settings().adjustCampaignId)
         }
 
-        SearchDefaultEngine.apply {
-            val defaultEngine = context
-                .components
-                .search
-                .searchEngineManager
-                .defaultSearchEngine ?: return@apply // TODO is it ok for this to be synchronous? it's already handling a null case...
+        scope.launch {
+            SearchDefaultEngine.apply {
+                val defaultEngine = context
+                    .components
+                    .search
+                    .searchEngineManager
+                    .getDefaultSearchEngineAsync(context)
 
-            code.set(defaultEngine.identifier)
-            name.set(defaultEngine.name)
-            submissionUrl.set(defaultEngine.buildSearchUrl(""))
-        }
+                code.set(defaultEngine.identifier)
+                name.set(defaultEngine.name)
+                submissionUrl.set(defaultEngine.buildSearchUrl(""))
```

Comment:
As far as I can tell, these properties are only used to send this ping, so making this async shouldn't change any behavior.